RecCalorimeter/src/components/FilterTruthParticles.cpp

```diff
@@ -0,0 +1,46 @@
+#include "FilterTruthParticles.h"
+
+// std
+#include <vector>
+#include <math.h>
+
+
+DECLARE_COMPONENT(FilterTruthParticles)
+
+FilterTruthParticles::FilterTruthParticles(const std::string& name, ISvcLocator* svcLoc) : Transformer(name, svcLoc,
+                    KeyValue("InputCollection", "MCParticles"),
+                    KeyValue("OutputCollection", "FilteredMCParticles")) {
+
+  declareProperty("SaveMuons", m_useMuons, "Bool for if muons are stored");
+  declareProperty("MinPt", m_minPt, "Minimum pT for filtered particles");
+}
+
+
+
+edm4hep::MCParticleCollection  FilterTruthParticles::operator()(const edm4hep::MCParticleCollection& input) const{
+
+  edm4hep::MCParticleCollection filteredParticles = edm4hep::MCParticleCollection();
+
+  for(auto particle: input){
+
+    // Only want to include final state particles (before GEANT simulation)
+    if(particle.isCreatedInSimulation()) continue;
+    if(particle.getGeneratorStatus() != 1) continue;
+
+    // Sometimes we want to ignore muons for jet reconstruction
+    if(m_useMuons && std::abs(particle.getPDG()) == 13) continue; 
```

Comment:
If you set `m_useMuons` to `true`, you actually skip muons in this logic, no?

Reply:
Good catch, I will flip that logic.